### PR TITLE
fix(tracing): fix context cloning

### PR DESCRIPTION
**Pull request by mabdinur**
Currently `Context._is_remote` is set to `True` on ALL child spans. This is incorrect. 

FIX: When copying a Context object from a parent span to a child span ensure is_remote is set to `False`!

This regression was introduced in: https://github.com/DataDog/dd-trace-py/commit/90a3e3fed47763abf64cb4c04c6a32f5c64076cb. This bug is unreleased and does not require a release note.

`isRemote` is an OpenTelemetry concept and it's expected behavior is described here: https://opentelemetry.io/docs/specs/otel/trace/api/#isremote

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/tracer/test_context.py [hunk 1]
```diff
@@ -363,3 +363,21 @@ def test_tracestate(context, expected_tracestate):
...
+
+
+def test_is_remote():
+    # type: () -> None
+    """Ensure that the is_remote flag is set to False on all local spans"""
+    # Context._is_remote should be True by default
+    ctx = Context(trace_id=123, span_id=321)
+    assert ctx._is_remote is True
+
+    # is_remote should be set to False in the context of a local span
```
Comment (mabdinur) [inline tests/tracer/test_context.py:375]:
```suggestion
    # span.context.is_remote should ALWAYS evaluate to False.
```

File: tests/tracer/test_context.py [hunk 2]
```diff
@@ -363,3 +363,21 @@ def test_tracestate(context, expected_tracestate):
...
+    local_span = Span("span_with_context", context=ctx)
+    local_span.context.trace_id = 123
+    local_span.context.span_id = 321
+    local_span.context._is_remote = False
+
+    # is_remote should be False on root spans
```
Comment (mabdinur) [inline tests/tracer/test_context.py:381]:
```suggestion
    # is_remote should be set to False on root spans
```

File: tests/tracer/test_context.py [hunk 3]
```diff
@@ -86,8 +86,8 @@ def validate_traceparent(context, sampled_expected):
     "context",
     [
         Context(),
-        Context(trace_id=123, span_id=321),
-        Context(trace_id=123, span_id=321, dd_origin="synthetics", sampling_priority=2),
+        Context(trace_id=123, span_id=321, is_remote=True),
```
Comment (erikayasuda) [inline tests/tracer/test_context.py:89]: Isn't `is_remote=True` the default? Or is this just for clarity

**Review by erikayasuda — APPROVED** [review]
Just one clarifying Q, otherwise LGTM

File: tests/tracer/test_context.py (hunk 3 above)
Comment (mabdinur) [reply tests/tracer/test_context.py:89]: Yup, it's true by default. I just wanted to test all the different ways the Context object could be initialized.

**Pull request merged**